diff --git a/node_modules/@maka/ui/node_modules/virtua/lib/index.js b/node_modules/@maka/ui/node_modules/virtua/lib/index.js
--- a/node_modules/@maka/ui/node_modules/virtua/lib/index.js
+++ b/node_modules/@maka/ui/node_modules/virtua/lib/index.js
@@ -49,7 +49,7 @@
         o: () => $,
         O: () => m,
         t: s,
-        L: () => (_ = w, w = 0, [ _, 2 === b ]),
+        L: () => { const pending = w; return _ += pending, w = 0, [ pending, 2 === b, _ ]; },
         H: (e, t) => {
             const o = [ e, t ];
             return T.add(o), () => {
@@ -218,8 +218,8 @@
             t.removeEventListener("touchend", w), $(s);
         },
         A: () => {
-            const [t, o] = e.L();
-            t && (r(t, o, a), a = !1, o && e.o() > e.t() && e.W(1, n()));
+            const [t, o, total] = e.L();
+            t && (r(total, o, a), a = !1, o && e.o() > e.t() && e.W(1, n()));
         }
     };
 }, H = (e, t, o) => {
diff --git a/node_modules/@maka/ui/node_modules/virtua/lib/index.cjs b/node_modules/@maka/ui/node_modules/virtua/lib/index.cjs
--- a/node_modules/@maka/ui/node_modules/virtua/lib/index.cjs
+++ b/node_modules/@maka/ui/node_modules/virtua/lib/index.cjs
@@ -67,7 +67,7 @@
         o: () => p,
         T: () => $,
         t: s,
-        j: () => (_ = w, w = 0, [ _, 2 === I ]),
+        j: () => { const pending = w; return _ += pending, w = 0, [ pending, 2 === I, _ ]; },
         L: (e, t) => {
             const o = [ e, t ];
             return C.add(o), () => {
@@ -236,8 +236,8 @@
             t.removeEventListener("touchend", w), $(s);
         },
         Y: () => {
-            const [t, o] = e.j();
-            t && (r(t, o, a), a = !1, o && e.o() > e.t() && e.q(1, n()));
+            const [t, o, total] = e.j();
+            t && (r(total, o, a), a = !1, o && e.o() > e.t() && e.q(1, n()));
         }
     };
 }, q = (e, t, o) => {
