diff --git a/node_modules/virtua/lib/index.cjs b/node_modules/virtua/lib/index.cjs
index 0153fca..7481c9e 100644
--- a/node_modules/virtua/lib/index.cjs
+++ b/node_modules/virtua/lib/index.cjs
@@ -361,8 +361,8 @@ n), S = e => l(e.t(), e.o()), _ = ({l: e, i: t, u: o, h: n, $: r, p: c, t: a, m:
             const [t, n] = e.L();
             if (!t) return;
             p && s && s(), p = !1;
-            const r = e.M() + t;
-            r <= 0 || r >= e.O() + e.t() - e.o() ? y(r) : o.scrollBy({
+            const r = e.M() + t, edge = e.O() + e.t() - e.o();
+            r <= 0 || r >= edge || z() >= edge - 1 ? y(r) : o.scrollBy({
                 [_]: v(t),
                 behavior: "instant"
             }), n && (i && i(), e.o() > e.t() && e.A(1, z()));
diff --git a/node_modules/virtua/lib/index.js b/node_modules/virtua/lib/index.js
index eac48a4..935ccfa 100644
--- a/node_modules/virtua/lib/index.js
+++ b/node_modules/virtua/lib/index.js
@@ -361,8 +361,8 @@ f), b = e => h(e.t(), e.o()), k = ({l: e, i: t, u: o, h: n, $: r, p: s, t: l, m:
             const [t, n] = e.L();
             if (!t) return;
             h && s && s(), h = !1;
-            const r = e.M() + t;
-            r <= 0 || r >= e.O() + e.t() - e.o() ? z(r) : o.scrollBy({
+            const r = e.M() + t, edge = e.O() + e.t() - e.o();
+            r <= 0 || r >= edge || v() >= edge - 1 ? z(r) : o.scrollBy({
                 [p]: m(t),
                 behavior: "instant"
             }), n && (i && i(), e.o() > e.t() && e.W(1, v()));
